src/formatDate.js

```diff
@@ -9,6 +9,34 @@
  */
 function formatDate(date, fromFormat, toFormat) {
   // write code here
+  const seperator = date.split(fromFormat[3]);
+  let formattedDate = [];
+  const format = {};
+
+  for (let i = 0; i < fromFormat.length - 1; i++) {
+    format[fromFormat[i]] = seperator[i];
+  }
+
+  for (let j = 0; j < toFormat.length; j++) {
+    if (format.hasOwnProperty(toFormat[j])) {
+      formattedDate.push(format[toFormat[j]]);
+    }
+
+    if (format.hasOwnProperty('YYYY') && toFormat[j] === 'YY') {
+      formattedDate.push(format['YYYY'].slice(2));
```

Comment:
The condition and the action for converting 'YYYY' to 'YY' can be moved outside of the loop since it's a one-time operation. This will make the code cleaner and avoid unnecessary checks in each iteration.